lib/puppet/module.rb

```diff
@@ -428,22 +428,25 @@ def initialize_i18n
 
     locales_path = File.absolute_path('locales', path)
 
-    begin
-      Puppet::GettextConfig.initialize(locales_path, :po)
+    if Puppet::GettextConfig.initialize(locales_path, :po)
       Puppet.debug "#{module_name} initialized for i18n: #{GettextSetup.translation_repositories[module_name]}"
-    rescue
-      config_path = File.absolute_path('config.yaml', locales_path)
-      Puppet.debug "Could not find locales configuration file for #{module_name} at #{config_path}. Skipping i18n initialization."
+    else
+      if Puppet::GettextConfig.gettext_found?
```

Comment:
When we disable_i18n, we don't do anything about the value of gettest_found, which means when its disabled, we'll call GettestConfig.initialize() which will return nil and then we end up calling File.absolute_path() here even though we disabled it.

Should we either check both values or alter the value of gettext_found?

Reply:
What do you guys think about adding a check for gettext being disabled here: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/module.rb#L101 ?

Reply:
How about immediately return from initialize_i18n if gettext is disabled?

Reply:
I kind of prefer the first one, because I feel like it surfaces what is going on better. Why am I even calling a function called `initialize_i18n` if all i18n is disabled? But I could go either way. I do think we should add it though, to cut out as much unnecessary work as possible.

Reply:
I'm fine with that reasoning.